templates/raspbian/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -140,37 +140,47 @@ cattlepi_run_bootcode()
 
 cattlepi_fetch_update_images()
 {
+    # Read the md5's of the current init/rootfs images for the purpose
+    # of removing them after updates
+    if [[ -r "${CATTLEPI_DIR}/config" ]]; then
+        local cur_initfs_md5=$(jq '.initfs.md5sum' "${CATTLEPI_DIR}/config")
+        local cur_rootfs_md5=$(jq '.rootfs.md5sum' "${CATTLEPI_DIR}/config")
+    fi
+
     cattlepi_fetch_update_config
+
     for img in initfs rootfs; do
         local md5sum=$(cat "${CATTLEPI_DIR}/config" | jq -r .${img}.md5sum )
-        if [ -r "${CATTLEPI_DIR}/${md5sum}" ]; then        
-            if [ "${CATTLEPI_MD5SKIP}" != "yes" ]; then 
+        if [ -r "${CATTLEPI_DIR}/${md5sum}" ]; then
+            if [ "${CATTLEPI_MD5SKIP}" != "yes" ]; then
                 echo "${md5sum}  ${CATTLEPI_DIR}/${md5sum}" > /tmp/check_md5
                 md5sum --quiet -c /tmp/check_md5 || rm -rf ${CATTLEPI_DIR}/${md5sum}
             fi
         fi
-        if [ ! -r "${CATTLEPI_DIR}/${md5sum}" ]; then 
+        if [ ! -r "${CATTLEPI_DIR}/${md5sum}" ]; then
             local filetodownload=$(cat "${CATTLEPI_DIR}/config" | jq -r .${img}.url )
             cattlepi_absolute_download $filetodownload "${CATTLEPI_DIR}/${md5sum}"
             if [ "${CATTLEPI_MD5SKIP}" != "yes" ]; then
                 echo "${md5sum}  ${CATTLEPI_DIR}/${md5sum}" > /tmp/check_md5
                 md5sum --quiet -c /tmp/check_md5 || panic "checksum verification failed"
             fi
+            # Delete the (old) images to save some space on the SD card
+            rm -rf "${cur_initfs_md5}" "${cur_rootfs_md5}"
```

Comment:
this is better than what we have today but has the potential to leak images

Reply:
does this work if only one of the images changes? will you just blow away both current images?